etna/lib/etna/clients/magma/workflows/walk_model_tree_workflow.rb

```diff
@@ -0,0 +1,101 @@
+require 'ostruct'
+require 'digest'
+require 'fileutils'
+require 'tempfile'
+
+module Etna
+  module Clients
+    class Magma
+      class WalkModelTreeWorkflow < Struct.new(:magma_crud, :logger, keyword_init: true)
```

Comment:
I suspect this workflow will be useful for other purposes, too.  It's a general magma walker that could be helped to also do, say, magma table joins